### PR TITLE
`unliftio-core`: Add `liftIOOp`

### LINKED ISSUE #108: Add `unlift :: MonadUnliftIO m => (IO a -> IO b) -> m a -> m b`?
What do you think about adding 

```haskell
    unlift :: MonadUnliftIO m => (IO a -> IO b) -> m a -> m b
    unlift m x = do
      runInIO <- askRunInIO
      liftIO $ m $ runInIO x
```

to `Control.Monad.IO.Unlift`? This is handy when you have a `IO a -> IO b`, and avoids a little bit of boilerplate. 

Usage example:

```haskell
runSTControl lock stControl prepare =
  maybe (throwIO Locked) return =<< unlift (withLockTry lock) do
```

where [`withLockTry`](https://hackage.haskell.org/package/extra-1.7.12/docs/Control-Concurrent-Extra.html#v:withLockTry) is `withLockTry :: Lock -> IO a -> IO (Maybe a)` from `extra`.

### PR DESCRIPTION
**Pull request by ocharles**
Fixes #108

### CODE REVIEW

**Comment by ocharles** [comment]
I'm not really sure about this name anymore. "Unlifting" is more about going _to_ `IO`, while this functions goes both to `IO` _and back_.

**Comment by snoyberg** [comment]
monad-control calls a similarly-typed function [`liftBaseOp_`](https://www.stackage.org/haddock/nightly-2023-01-30/monad-control-1.0.3.1/Control-Monad-Trans-Control.html#v:liftBaseOp_). Maybe that makes sense?

**Comment by ocharles** [comment]
Ah yes! I was looking in `monad-control` but missed that one. Maybe we go with `liftIOOp`?

**Comment by snoyberg** [comment]
Sounds good to me

**Comment by ocharles** [comment]
@snoyberg Rewritten, I think this is good to go.

File: unliftio-core/src/Control/Monad/IO/Unlift.hs [hunk 1]
```diff
@@ -160,3 +161,15 @@ wrappedWithRunInIO :: MonadUnliftIO n
...
 wrappedWithRunInIO wrap unwrap inner = wrap $ withRunInIO $ \run ->
   inner $ run . unwrap
+
+{- | A helper function for lifting @IO a -> IO b@ functions into any @MonadUnliftIO@.
+
+=== __Example__
+
+> liftedTry :: (Exception e, MonadUnliftIO m) => m a -> m (Either e a)
+> liftedTry m = liftIOOp Control.Exception.try m
+-}
```
Comment (snoyberg) [inline unliftio-core/src/Control/Monad/IO/Unlift.hs:171]: Can you add a minor version bump, since comment, and changelog entry?

Comment (snoyberg) [reply unliftio-core/src/Control/Monad/IO/Unlift.hs:171]: Or rather: update changelog with the new minor version

Comment (ocharles) [reply unliftio-core/src/Control/Monad/IO/Unlift.hs:171]: Done :ok_hand:

**Review by snoyberg — APPROVED** [review]
Thanks!

**Pull request merged**